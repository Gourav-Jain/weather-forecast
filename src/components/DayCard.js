import React from 'react';
import { getFormatedTime, API_IMG_URL } from '../utils';

export const DayCard = ({ forecastData }) => {
    return (
        <>
            {forecastData.map((val, key) => (
                <div key={key} className="days-container">
                    <div className="container">
                        <h3 className="date">{val.date}</h3>
                        <div className="weather-icon">
                            <div className="sunny">
                                <img alt="" src={`${API_IMG_URL}${val.forecast[0].weather[0].icon}@2x.png`} />
                            </div>
                            <p className="temp">{val.forecast[0].main.temp} <span>&#8451;</span></p>
                            <p className="conditions">{val.forecast[0].weather[0].description}</p>
                        </div>
                        <div className="time-wrapper">
                            <div className="time-wrap">
                                    <div className="text">
                                        <p>Time:</p>
                                        <p>Min Temp:</p>
                                        <p>Max Temp:</p>
                                    </div>
                                {val.forecast.map((itm, key) => (
                                    <div key={key} className="text">
                                        <p>{`${getFormatedTime(itm.dt_txt)}`}</p>
                                        <p>{itm.main.temp_min} <span>&#8451;</span></p>
                                        <p>{itm.main.temp_max} <span>&#8451;</span></p>
                                    </div>
                                ))}
                  
                            </div>
                        </div>
                    </div>
                </div>
            ))}
        </>
    )
}

export default DayCard;