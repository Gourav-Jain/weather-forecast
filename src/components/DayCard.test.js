import React from 'react';
import { render } from '@testing-library/react';
import DayCard from './DayCard';


const data = [{ "date": "Saturday, September 5, 2020", "forecast": [{ "dt": 1599328800, "main": { "temp": 15.12, "temp_min": 13.34, "temp_max": 15.12 }, "weather": [{ "id": 500, "main": "Rain", "description": "light rain", "icon": "10d" }], "dt_txt": "2020-09-05 18:00:00" }, { "dt": 1599339600, "main": { "temp": 11.17, "temp_min": 9.61, "temp_max": 11.17,}, "weather": [{ "id": 801, "main": "Clouds", "description": "few clouds", "icon": "02n" }], "dt_txt": "2020-09-05 21:00:00" }] }]

test('@DayCard', () => {
  const { container } = render(<DayCard forecastData={data} />);
  expect(container).toMatchSnapshot()
});