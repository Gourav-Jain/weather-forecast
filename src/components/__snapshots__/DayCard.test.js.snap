// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`@DayCard 1`] = `
<div>
  <div
    class="days-container"
  >
    <div
      class="container"
    >
      <h3
        class="date"
      >
        Saturday, September 5, 2020
      </h3>
      <div
        class="weather-icon"
      >
        <div
          class="sunny"
        >
          <img
            alt=""
            src="http://openweathermap.org/img/wn/10d@2x.png"
          />
        </div>
        <p
          class="temp"
        >
          15.12
           
          <span>
            ℃
          </span>
        </p>
        <p
          class="conditions"
        >
          light rain
        </p>
      </div>
      <div
        class="time-wrapper"
      >
        <div
          class="time-wrap"
        >
          <div
            class="text"
          >
            <p>
              Time:
            </p>
            <p>
              Min Temp:
            </p>
            <p>
              Max Temp:
            </p>
          </div>
          <div
            class="text"
          >
            <p>
              18:00
            </p>
            <p>
              13.34
               
              <span>
                ℃
              </span>
            </p>
            <p>
              15.12
               
              <span>
                ℃
              </span>
            </p>
          </div>
          <div
            class="text"
          >
            <p>
              21:00
            </p>
            <p>
              9.61
               
              <span>
                ℃
              </span>
            </p>
            <p>
              11.17
               
              <span>
                ℃
              </span>
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
